compiler/rustc_errors/src/emitter.rs

```diff
@@ -1713,7 +1713,15 @@ impl EmitterWriter {
         let max_line_num_len = if self.ui_testing {
             ANONYMIZED_LINE_NUM.len()
         } else {
-            self.get_max_line_num(span, children).to_string().len()
+            let mut n = self.get_max_line_num(span, children);
```

Comment:
Could you add a comment explaining why we calculate the number of digits this way? (To make sure someone doesn't revert this change without realising it's intentional, later.)

Reply:
Added an explanatory comment.